synapse/handlers/message.py

```diff
@@ -1036,6 +1058,9 @@ async def cache_joined_hosts_for_event(self, event: EventBase) -> None:
                 expiry_ms=60 * 60 * 1000,
             )
 
+            if self._external_cache_joined_hosts_updates:
+                self._external_cache_joined_hosts_updates[context.prev_group] = None
```

Comment:
why is this `context.prev_group` rather than `state_entry.state_group` ?

Reply:
They're (almost certainly) the same value, but we should use `state_entry.state_group`